packages/react/src/components/Tabs/Tabs.tsx

```diff
@@ -85,17 +74,28 @@ const Tabs = ({
   };
 
   const tabComponents = tabs.map((child, index) => {
-    const { ...other } = (child as React.ReactElement<any>).props;
+    const { target, id: propId, ...other } = (child as React.ReactElement<
+      any
+    >).props;
     const selected = index === activeIndex;
+    const [id] = propId ? [propId] : useId(1, 'tab');
+
+    useEffect(() => {
+      target.current?.setAttribute('aria-labelledby', id);
+    }, [target]);
+
+    useEffect(() => {
+      index === activeIndex
+        ? target.current?.classList.remove('TabPanel--hidden')
+        : target.current?.classList.add('TabPanel--hidden');
+    }, [activeIndex]);
 
     const config = {
-      id: `${id}-${index}`,
```

Comment:
Still need to set id here. Right now, it's only set if the user has supplied it. It should be safe just to set id because it's set above:

```jsx
const [id] = propId ? [propId] : useId(1, 'tab')
```